pkg/controllers/management/nodepool/nodepool.go

```diff
@@ -140,15 +158,32 @@ func (c *Controller) Remove(nodePool *v3.NodePool) (runtime.Object, error) {
 			continue
 		}
 
-		if err := c.deleteNode(&node, 0); err != nil {
+		if err := c.deleteNodeBackoffAndRetry(&node); err != nil {
 			return nodePool, err
 		}
 	}
 
 	return nodePool, nil
 }
 
-func (c *Controller) setReconcileAnnotation(nodePool *v3.NodePool, anno string) (*v3.NodePool, error) {
+// nodes returns a slice of node objects listed directly from the Kubernetes API.
+func (c *Controller) nodes(nodePool *v3.NodePool) ([]*v3.Node, error) {
+	nodeList, err := c.Nodes.ListNamespaced(nodePool.Namespace, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var nodes []*v3.Node
```

Comment:
Do we need to create a new list here? since it's a new func, signature could be []*v3.NodeList and loop over list.Items in the logic